scripts/deploy.py

```diff
@@ -43,18 +43,68 @@ def upload_archive(archive, zenodo_json, version):
     headers = {"Accept": "application/json"}
     params = {"access_token": ZENODO_TOKEN}
 
-    # Create an empty upload
-    response = requests.post(
-        "https://zenodo.org/api/deposit/depositions",
-        params=params,
-        json={},
-        headers=headers,
-    )
-    if response.status_code not in [200, 201]:
-        sys.exit(
-            "Trouble requesting new upload: %s, %s"
-            % (response.status_code, response.json())
+    if doi:
```

Comment:
Since we now have two giant blocks of spaghetti code it might make sense to either refactor this into a class (and then call much shorter functions for each block depending on the action) or to make separate functions. We want to eliminate redundancy for clarity! I can also help with this if you are at your point of feeling overwhelmed or tired

Reply:
:laughing: Not overwhelmed, but happy to pass it off if you're better with python idioms (I'm not).

Reply:
Yep happy to take on that part!